packages/autocomplete-core/src/__tests__/getSources.test.ts

```diff
@@ -37,7 +41,88 @@ describe('getSources', () => {
     });
   });
 
-  test.todo('provides default source implementations');
+  test('provides a default source implementations', async () => {
+    const onStateChange = jest.fn();
+    const { inputElement } = createPlayground(createAutocomplete, {
+      getSources: () => {
+        return [
+          {
+            getItems() {
+              return [];
+            },
+            templates: {
+              item() {},
+            },
+          },
+        ];
+      },
+      onStateChange,
+    });
+
+    inputElement.focus();
+    userEvent.type(inputElement, 'a');
+
+    await runAllMicroTasks();
+
+    expect(onStateChange).toHaveBeenCalledWith({
+      prevState: expect.anything(),
+      state: expect.objectContaining({
+        collections: expect.arrayContaining([
+          expect.objectContaining({
+            source: expect.objectContaining({
+              getItemInputValue: expect.any(Function),
+              getItemUrl: expect.any(Function),
+              getItems: expect.any(Function),
+              onActive: expect.any(Function),
+              onSelect: expect.any(Function),
+              templates: expect.objectContaining({
+                item: expect.any(Function),
+              }),
+            }),
```

Comment:
This is more strict and should work, right?

```suggestion
            source: {
              getItemInputValue: expect.any(Function),
              getItemUrl: expect.any(Function),
              getItems: expect.any(Function),
              onActive: expect.any(Function),
              onSelect: expect.any(Function),
              templates: expect.objectContaining({
                item: expect.any(Function),
              }),
            },
```

Reply:
It is indeed and it does work!